### PR TITLE
feat(dir): recurse among files

### PR DESCRIPTION
**Pull request by cgestes**
This is a started commit. It uses ignore from ripgrep.

so it respects .gitignore and ignores hidden files

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -1,3 +1,40 @@
...
+use clap::{App, Arg};
+use ignore::Walk;
+
 fn main() {
-    println!("Hello, world!");
+    let matches = App::new("fr")
+        .version("0.1.0")
+        .author("The ruplacer team")
+        .about("Find and replace")
+        .arg(Arg::with_name("PATERN")
```
Comment (blastrock) [inline src/main.rs]: *PATTERN

File: src/main.rs [hunk 2]
```diff
@@ -1,3 +1,40 @@
...
+            .index(1))
+        .arg(Arg::with_name("DIR")
+            .short("d")
+            .long("dir")
+            .value_name("DIR")
+            .index(2)
+            .help("the directory to find into")
+            .takes_value(true))
+        .arg(Arg::with_name("v")
+            .short("v")
```
Comment (blastrock) [inline src/main.rs]: can't we have a --verbose with short -v?

File: src/main.rs (hunk 1 above)
Comment (cgestes) [reply src/main.rs]: done

File: src/main.rs (hunk 2 above)
Comment (cgestes) [reply src/main.rs]: done

**Comment by cgestes** [comment]
superseeded by #3

**Pull request closed without merging**